h2/stream.py

```diff
@@ -1040,6 +1040,9 @@ def _build_hdr_validation_flags(self, events):
             is_trailer = isinstance(
                 events[0], (_TrailersSent, TrailersReceived)
             )
+            is_response_header = isinstance(
+                events[0], ResponseReceived
+            ) and not self.state_machine.client
```

Comment:
Do we need this `and not` check? I believe that `ResponseReceived` can only fire when we are a client.

While we're here, I suspect we're going to need an internal-only `_ResponseSent` event, much like the `_TrailersSent` event you can see in the block above.